universum.cabal

```diff
@@ -143,10 +143,18 @@ test-suite universum-doctest
   main-is:             Doctest.hs
 
   build-depends:       doctest
+                       -- This is necessary as long as we support GHC-8.4.
+                       -- Cabal is a boot library and in case of GHC-8.2 it has version 2.2.
+                       -- It causes troubles to `ghc-paths` whose `Setup.hs` has a
+                       -- compile-time conditional that depends on the Cabal version.
+                       -- If Cabal is older than 2.3.0, it will use some deprecated function
+                       -- which causes compilation errors with modern tools.
+                     , Cabal >= 2.4
```

Comment:
This version of Cabal was released not so long ago (September 2018), maybe try to require it only if GHC version is high enough?

Reply:
@Martoon-00 `Cabal` is a boot library and GHC versions ≥ 8.6 use Cabal ≥ 2.4 according to [this](https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/libraries/version-history).

So for GHC ≥ 8.6 this constraint is sort of implicitly required. And for GHC-8.4 we manually force it for the reasons described in my comment. However, I can't reproduce the problem which forced me to require that constraint anymore. Apparently CI passes even without that commit. I am not sure why, maybe some packages were updated since then and now a slightly different build plan is used :shrug: So I believe this concern is fully resolved (I simply dropped that commit).